src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -1014,6 +1016,7 @@ impl WithOptionName for KafkaSinkConfigOptionName {
             | KafkaSinkConfigOptionName::TopicConfig
             | KafkaSinkConfigOptionName::TopicPartitionCount
             | KafkaSinkConfigOptionName::TopicReplicationFactor => false,
+            KafkaSinkConfigOptionName::PartitionBy => true,
```

Comment:
why do we want to redact this expression?

Reply:
The expression could contain string or numeric literals, and per our data management policy any string literals need to be redacted.